MSTR-114: Add markdown editor (#533)
2600hz/monster-ui: docs/monster-ui/markdownEditor().md

title: markdownEditor()

# monster.ui.markdownEditor()

> The `monster.ui.markdownEditor()` function convert a simple textarea into a markdown editor.

Reviewer: ```suggestion
The `monster.ui.markdownEditor()` method converts a simple textarea into a markdown editor.
```

![](images/markdownEditor-preview.png)

## Syntax
```javascript
monster.ui.markdownEditor(target[, options]);
```

### Parameters

Key | Description | Type | Default | Required
:-: | --- | :-: | :-: | :-:

> `target` | A jQuery object that will be converted to markdow editor. | `jQuery` | | `true`

Reviewer: You should avoid writing in the future tense when describing parameters in general (as it makes it unnecessarily verbose).
```suggestion
`target` | A jQuery object to be converted into a markdown editor. | `jQuery` | | `true`
```

> `options` | This will be a JavaScript Object, see the plugin [markdownEditor docs](https://simplemde.com/) for reference. | `Object`([#/options](#options)) | | `false`

Reviewer: ```suggestion
`options` | List of options used to instantiate a markdown editor (available [options](https://github.com/sparksuite/simplemde-markdown-editor#configuration)). | `Object`([#/options](#options)) | | `false`
```

## Description

> The `monster.ui.markdownEditor()` method convert a simple textarea element into a powerful markdown editor which is specified by the `target` parameter. The toolbar can be customized or hidden completely, please see [markdownEditor docs](https://simplemde.com/) for reference.

Reviewer: ```suggestion
The `monster.ui.markdownEditor()` method converts a simple textarea element specified by the `target` parameter into a powerful markdown editor. The toolbar can be customized or hidden completely, please see [markdownEditor docs](https://simplemde.com/) for reference.
```
